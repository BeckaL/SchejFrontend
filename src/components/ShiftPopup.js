import React, { Component } from "react";
import RequestSwapForm from "./RequestSwapForm";
import EmergencyRequestForm from "./EmergencyRequestForm";

export default class ShiftPopup extends Component {
  constructor(props) {
    super(props)
    this.state = {
      formatStart: '',
      formatEnd: '',
      showingRequestForm: false,
      showingEmergencyForm: false
    }
  }
  componentWillMount() {
    this.setState({
      formatStart: this.props.shiftInfo.start.toLocaleString("en-GB"),
      formatEnd: this.props.shiftInfo.end.toLocaleString("en-GB")
    })
  }

  changeState = () => {
    this.setState({
      showingRequestForm: true
    })
  }

  emergencyRequest = () => {
    this.setState({
      showingEmergencyForm: true
    })
  }

  formatEmergencyRequest = () => {
    if (this.props.shiftInfo.userId === parseInt(localStorage['id'])) {
      return(
        <button id='toggle-emergency-popup-content' className='custom-button' onClick={this.emergencyRequest}>Request emergency cover</button>
    )}
  }

  render() {
    if (this.state.showingRequestForm) {
      return (
        <div id='request-swap'>
          <h3 className='popup-title'>Request a shift swap</h3>
          <RequestSwapForm
            shiftInfo = {this.props.shiftInfo}
            history={this.props.history}
          />
        </div>
      )
    } else if (this.state.showingEmergencyForm) {
        return (
          <div id='emergency-request'>
            <h3 className='popup-title'>Emergency Request Form</h3>
            <EmergencyRequestForm
              shiftInfo = {this.props.shiftInfo}
              history={this.props.history}
            />
          </div>
        )
    } else {
      return(
        <div id='shift-popup'>
          <h3 className='popup-title'>Shift Information</h3>
          <div className='shift-info'>
            <p className='shift-info-label'>Employee name:</p>
            <p className='shift-info-entry'>{this.props.shiftInfo.title}</p>
          </div>
          <br />
          <div className='shift-info'>
            <p className='shift-info-label'>Contact email address:</p>
            <p className='shift-info-entry'>{this.props.shiftInfo.email}</p>
          </div>
          <br />
          <div className='shift-info'>
            <p className='shift-info-label'>Start time:</p>
            <p className='shift-info-entry'>{this.state.formatStart}</p>
          </div>
          <br />
          <div className='shift-info'>
            <p className='shift-info-label'>End time:</p>
            <p className='shift-info-entry'>{this.state.formatEnd}</p>
          </div>
          <br />
          <section id='toggle-popup-state'>
            <button id='toggle-popup-content' className='custom-button' onClick={this.changeState}>Request shift swap</button>
            {this.formatEmergencyRequest()}
          </section>
        </div>
      )}
    }
}
