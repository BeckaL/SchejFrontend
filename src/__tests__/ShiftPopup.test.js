import React from 'react'
import { mount, shallow } from 'enzyme'
import ShiftPopup from '../components/ShiftPopup'

describe('Shift Popup', () => {
  var wrapper

  beforeAll(() => {
    wrapper = shallow(<ShiftPopup shiftInfo = {
      {
        start: { toLocaleString() {} },
        end: { toLocaleString() {} }
      }
    }/>)
  })


  test("it matches the snapshot", () => {
    console.log(wrapper);
    expect(wrapper).toMatchSnapshot()
  })

  test('it renders the shift popup div ', () => {
    expect(wrapper.find('#shift-popup')).toBeTruthy()
  })

  

  })

})
