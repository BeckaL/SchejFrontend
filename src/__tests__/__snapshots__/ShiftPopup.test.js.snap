// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shift Popup it matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShiftPopup
    shiftInfo={
      Object {
        "end": Object {
          "toLocaleString": [Function],
        },
        "start": Object {
          "toLocaleString": [Function],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="popup-title"
        >
          Shift Information
        </h3>,
        <div
          className="shift-info"
        >
          <p
            className="shift-info-label"
          >
            Employee name:
          </p>
          <p
            className="shift-info-entry"
          />
        </div>,
        <br />,
        <div
          className="shift-info"
        >
          <p
            className="shift-info-label"
          >
            Contact email address:
          </p>
          <p
            className="shift-info-entry"
          />
        </div>,
        <br />,
        <div
          className="shift-info"
        >
          <p
            className="shift-info-label"
          >
            Start time:
          </p>
          <p
            className="shift-info-entry"
          />
        </div>,
        <br />,
        <div
          className="shift-info"
        >
          <p
            className="shift-info-label"
          >
            End time:
          </p>
          <p
            className="shift-info-entry"
          />
        </div>,
        <br />,
        <section
          id="toggle-popup-state"
        >
          <button
            className="custom-button"
            id="toggle-popup-content"
            onClick={[Function]}
          >
            Request shift swap
          </button>
        </section>,
      ],
      "id": "shift-popup",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Shift Information",
          "className": "popup-title",
        },
        "ref": null,
        "rendered": "Shift Information",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="shift-info-label"
            >
              Employee name:
            </p>,
            <p
              className="shift-info-entry"
            />,
          ],
          "className": "shift-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Employee name:",
              "className": "shift-info-label",
            },
            "ref": null,
            "rendered": "Employee name:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "shift-info-entry",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="shift-info-label"
            >
              Contact email address:
            </p>,
            <p
              className="shift-info-entry"
            />,
          ],
          "className": "shift-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Contact email address:",
              "className": "shift-info-label",
            },
            "ref": null,
            "rendered": "Contact email address:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "shift-info-entry",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="shift-info-label"
            >
              Start time:
            </p>,
            <p
              className="shift-info-entry"
            />,
          ],
          "className": "shift-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Start time:",
              "className": "shift-info-label",
            },
            "ref": null,
            "rendered": "Start time:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "shift-info-entry",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="shift-info-label"
            >
              End time:
            </p>,
            <p
              className="shift-info-entry"
            />,
          ],
          "className": "shift-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "End time:",
              "className": "shift-info-label",
            },
            "ref": null,
            "rendered": "End time:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "shift-info-entry",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="custom-button"
            id="toggle-popup-content"
            onClick={[Function]}
          >
            Request shift swap
          </button>,
          "id": "toggle-popup-state",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Request shift swap",
            "className": "custom-button",
            "id": "toggle-popup-content",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Request shift swap",
          "type": "button",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="popup-title"
          >
            Shift Information
          </h3>,
          <div
            className="shift-info"
          >
            <p
              className="shift-info-label"
            >
              Employee name:
            </p>
            <p
              className="shift-info-entry"
            />
          </div>,
          <br />,
          <div
            className="shift-info"
          >
            <p
              className="shift-info-label"
            >
              Contact email address:
            </p>
            <p
              className="shift-info-entry"
            />
          </div>,
          <br />,
          <div
            className="shift-info"
          >
            <p
              className="shift-info-label"
            >
              Start time:
            </p>
            <p
              className="shift-info-entry"
            />
          </div>,
          <br />,
          <div
            className="shift-info"
          >
            <p
              className="shift-info-label"
            >
              End time:
            </p>
            <p
              className="shift-info-entry"
            />
          </div>,
          <br />,
          <section
            id="toggle-popup-state"
          >
            <button
              className="custom-button"
              id="toggle-popup-content"
              onClick={[Function]}
            >
              Request shift swap
            </button>
          </section>,
        ],
        "id": "shift-popup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Shift Information",
            "className": "popup-title",
          },
          "ref": null,
          "rendered": "Shift Information",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="shift-info-label"
              >
                Employee name:
              </p>,
              <p
                className="shift-info-entry"
              />,
            ],
            "className": "shift-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Employee name:",
                "className": "shift-info-label",
              },
              "ref": null,
              "rendered": "Employee name:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "shift-info-entry",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="shift-info-label"
              >
                Contact email address:
              </p>,
              <p
                className="shift-info-entry"
              />,
            ],
            "className": "shift-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Contact email address:",
                "className": "shift-info-label",
              },
              "ref": null,
              "rendered": "Contact email address:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "shift-info-entry",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="shift-info-label"
              >
                Start time:
              </p>,
              <p
                className="shift-info-entry"
              />,
            ],
            "className": "shift-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Start time:",
                "className": "shift-info-label",
              },
              "ref": null,
              "rendered": "Start time:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "shift-info-entry",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="shift-info-label"
              >
                End time:
              </p>,
              <p
                className="shift-info-entry"
              />,
            ],
            "className": "shift-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "End time:",
                "className": "shift-info-label",
              },
              "ref": null,
              "rendered": "End time:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "shift-info-entry",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="custom-button"
              id="toggle-popup-content"
              onClick={[Function]}
            >
              Request shift swap
            </button>,
            "id": "toggle-popup-state",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Request shift swap",
              "className": "custom-button",
              "id": "toggle-popup-content",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Request shift swap",
            "type": "button",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
